include/swift/Remote/MetadataReader.h

```diff
@@ -260,7 +261,58 @@ class MetadataReader {
 
     return start;
   }
-  
+
+  /// Given a known-opaque existential, attemp to discover the pointer to its
+  /// metadata address and its value.
+  llvm::Optional<std::pair<RemoteAddress, RemoteAddress>>
+  readMetadataAndValueOpaqueExistential(RemoteAddress ExistentialAddress) {
+    // OpaqueExistentialContainer is the layout of an opaque existential.
+    // `Type` is the pointer to the metadata.
+    OpaqueExistentialContainer Container;
```

Comment:
This needs to be `TargetOpaqueExistentialContainer<Runtime>`.

...and that type needs to be properly target-ified, because it definitely is not right now.  (It's really bad that it's declared with the `Target<Runtime>` idiom but doesn't actually follow the rules correctly.)